hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/SemanticVersionComparator.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (C) 2022 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hedera.node.app.spi;
+
+import com.hederahashgraph.api.proto.java.SemanticVersion;
+import java.util.Comparator;
+
+/**
+ * An implementation of {@link Comparator} for {@link SemanticVersion}s.
+ *
+ * <p>NOTE: When PBJ automatically generates {@link SemanticVersion} as {@link Comparable} then we
+ * can remove this class.
+ */
+public class SemanticVersionComparator implements Comparator<SemanticVersion> {
```

Comment:
This is a very neat trick! I still would like to have a single instance of this comparator, since it is used from several places, and not copy/paste this code in several places. And, as the comments state, this will get replaced by generated code from PBJ at some point. But I will reduce this class to just be:

```
public class SemanticVersionComparator {
    public static final Comparator<SemanticVersion> INSTANCE =
            Comparator.comparingInt(SemanticVersion::getMajor)
            .thenComparingInt(SemanticVersion::getMinor)
            .thenComparingInt(SemanticVersion::getPatch);
}
```

Which is much nicer!